csrc/executor.cpp

```diff
@@ -1948,4 +1950,230 @@ float FusionExecutor::runRtc(
+  }
+
+  return serde::CreateFusionExecutorDirect(
+      builder,
+      device_smem_limit_,
+      block_size_high_water_mark_,
+      maxrregcount_high_water_mark_,
+      warp_size_,
+      fusion_id_,
+      fusion_id_counter_,
+      kernel_code_.c_str(),
+      &executor_entry_lookup_keys_fb,
+      &executor_entry_lookup_values_fb,
+      serde::mapToSerdeDtype(kernel()->indexType()));
+}
+
+flatbuffers::Offset<serde::ExecutorEntry> FusionExecutor::serialize(
+    flatbuffers::FlatBufferBuilder& builder,
+    const ExecutorEntry& data) const {
+  // See table definition for ExecutorEntry in serde/fusion_cache.fbs
+
+  // In the flatbuffer schema, we store the vector of pairs as a pair of
+  // vectors.
+  std::vector<int> output_aliases_fb;
+  std::vector<int> input_aliases_fb;
+  output_aliases_fb.reserve(data.output_to_input_aliases.size());
+  input_aliases_fb.reserve(data.output_to_input_aliases.size());
+  for (auto [out, in] : data.output_to_input_aliases) {
+    output_aliases_fb.push_back(out);
+    input_aliases_fb.push_back(in);
+  }
+
+  // Serialize GlobalBufferInfo for outputs.
+  // We map the output TensorView pointer to its corresponding position in
+  // fusion outputs assuming that the output ordering is consistent.
+  using fb_global_buffer_info =
+      flatbuffers::Offset<nvfuser::serde::GlobalBufferInfo>;
+  std::vector<fb_global_buffer_info> outputs_fb;
+  outputs_fb.reserve(data.outputs.size());
+  for (const auto& buffer : data.outputs) {
+    auto tv_iter = std::find(
+        kernel()->outputs().cbegin(), kernel()->outputs().cend(), buffer.tv);
+    auto tv_position = (tv_iter == kernel()->outputs().cend())
+        ? -1
+        : std::distance(kernel()->outputs().cbegin(), tv_iter);
+    outputs_fb.push_back(
+        serialize(builder, buffer, tv_position, true /* is_fusion_output */));
+  }
+
+  // Serialize GlobalBufferInfo for intermediates.
+  // We map the intermediate TensorView pointer to its corresponding position in
+  // KernelSummary global allocations. We assume that the ordering is consistent
+  // between GpuLower objects with the same scheduled fusion.
+  std::vector<fb_global_buffer_info> intermediates_fb;
+  intermediates_fb.reserve(data.intermediates.size());
+  for (const auto& buffer : data.intermediates) {
+    auto match_tv_predicate = [buffer_tv = buffer.tv](const kir::Allocate* a) {
+      return a->buffer() == buffer_tv;
+    };
+    auto tv_iter = std::find_if(
+        kernel()->summary().global_allocations.cbegin(),
+        kernel()->summary().global_allocations.cend(),
+        match_tv_predicate);
+    auto tv_position =
+        (tv_iter == kernel()->summary().global_allocations.cend())
+        ? -1
+        : std::distance(
+              kernel()->summary().global_allocations.cbegin(), tv_iter);
+    intermediates_fb.push_back(
+        serialize(builder, buffer, tv_position, false /* is_fusion_output */));
+  }
+
+  return CreateExecutorEntryDirect(
```

Comment:
I'll add `serde::` for consistency.

IIUC, scope isn't necessary here since the `serde` namespace is nested within `nvfuser` namespace. It is necessary if the name is duplicated in both namespaces.